### DISCUSSION TITLE
No autoconnect serial on jdy-67 device

### DISCUSSION THREAD

**Discussion by NespaLa** [blueman-project/blueman#2583] [Q&A]
applet in log debug mode

shows it just connects audio:

blueman-applet 12.22.21 DEBUG    Base:67 _properties_changed: /org/bluez/hci1/dev_49_94_11_08_58_C2 {'Connected': True} [] <Device.Device object at 0x7f5f39981240 (blueman+bluez+Device+Device at 0x55800d192270)>
blueman-applet 12.22.23 INFO     PulseAudioUtils:337 __event_callback: 9 550
blueman-applet 12.22.23 DEBUG    PulseAudioProfile:111 on_pa_event: 9 550
blueman-applet 12.22.23 INFO     PulseAudioProfile:124 on_pa_event: card
blueman-applet 12.22.23 INFO     PulseAudioProfile:131 on_pa_event: add
blueman-applet 12.22.23 INFO     PulseAudioUtils:337 __event_callback: 25 550
blueman-applet 12.22.23 DEBUG    PulseAudioProfile:111 on_pa_event: 25 550
blueman-applet 12.22.23 INFO     PulseAudioProfile:124 on_pa_event: card
blueman-applet 12.22.23 INFO     PulseAudioProfile:126 on_pa_event: change
blueman-applet 12.22.23 INFO     PulseAudioUtils:337 __event_callback: 25 550
blueman-applet 12.22.23 DEBUG    PulseAudioProfile:111 on_pa_event: 25 550
blueman-applet 12.22.23 INFO     PulseAudioProfile:124 on_pa_event: card
blueman-applet 12.22.23 INFO     PulseAudioProfile:126 on_pa_event: change
blueman-applet 12.22.23 INFO     PulseAudioUtils:337 __event_callback: 25 550
blueman-applet 12.22.23 DEBUG    PulseAudioProfile:111 on_pa_event: 25 550
blueman-applet 12.22.23 INFO     PulseAudioProfile:124 on_pa_event: card
blueman-applet 12.22.23 INFO     PulseAudioProfile:126 on_pa_event: change
blueman-applet 12.22.53 INFO     Notification:242 do_g_signal: 2
blueman-applet 12.22.53 DEBUG    Notification:248 do_g_signal: The notification was dismissed by the user.

This is mouse click serial port connect:

blueman-applet 12.23.25 INFO     RecentConns:112 notify    : <Device.Device object at 0x7f5f39981240 (blueman+bluez+Device+Device at 0x55800d192270)> 00001101-0000-1000-8000-00805f9b34fb
rfcomm channel: 7
blueman-applet 12.23.25 INFO     SerialService:103 connect   : Starting rfcomm watcher as root

blueman-manager info:
49:94:11:08:58:C2
public
JDY-67-Audio
JDY-67-Audio
0x260414
0x0000
audio-card
yes
yes
no
no
yes
00001101-0000-1000-8000-00805f9b34fb Serial Port
0000110b-0000-1000-8000-00805f9b34fb Audio Sink
0000110c-0000-1000-8000-00805f9b34fb Remote Control Target
0000110d-0000-1000-8000-00805f9b34fb Advanced Audio
0000110e-0000-1000-8000-00805f9b34fb Remote Control
0000110f-0000-1000-8000-00805f9b34fb Video Conferencing
0000111e-0000-1000-8000-00805f9b34fb Handsfree
/org/bluez/hci1

So the device indicates Serial Port possibility, but what would it make auto-connect?

versions:
blueman-manager 2.4.2
bluez-5.79-1
kernel 6.12.6-1-default opensuse tumbleweed

**Comment by infirit** [comment]
If with "autoconnect" you mean the Connect menu entry in blueman manager then that is intentional. There are a few profiles that are defined as auto connect like input and in your case audio. Just connecting the serial profile (00001101-0000-1000-8000-00805f9b34fb Serial Port) is not very useful.

What blueman does when you pick to connect Serial is a lot more. It figures out the right rfcomm channel, creates the rfcomm device node so you can interact with and insures it remains connected through a watcher that opens the device. You can even let the SerialManager plugin run scripts on connect.

What specifically are you looking for that you are missing in blueman?

**Reply by NespaLa** [reply]
In this jdy-67 application, one always need both audio and a serial /dev/rfcommX device.  
The use case is simular to this design https://github.com/BG6LH/FT-81x-BT-DIGI-Adaptor a bt device to connect audio in / out and serial command to and from a ham radio device.

It would be fine if some config file or triggered script would do that automatically at connect/pair time.
A fixed device name would even be better, instead of a moving target /dev/rfcommX

If you have hints on how to do that in or outside Blueman, it would be more than welcome!

**Reply by infirit** [reply]
> If you have hints on how to do that in or outside Blueman, it would be more than welcome!

Officially it's complicated as the `rfcomm` command line utility has been deprecated by BlueZ. I don't do serial work so the quirks and gotchas may be lost on me but if you know the mac address and the channel number you can essentially do:

1. `bluetoothctl connect <mac>`
2. `rfcomm bind <mac> <channel>`
3. Do your serial thing on /dev/rfcommX

**Reply by NespaLa** [reply]
No clue on the channel, even after trying sdptools.

Blueman-manager always goes for 7

blueman-applet 17.39.17 INFO     RecentConns:112 notify    : <Device.Device object at 0x7f5f39981240 (blueman+bluez+Device+Device at 0x55800d192270)> 00001101-0000-1000-8000-00805f9b34fb
rfcomm channel: 7
blueman-applet 17.39.17 INFO     SerialService:103 connect   : Starting rfcomm watcher as root

Looking for an rfcomm replacement via a search engine, looks like 7 or 8 years confusion without any solution but blueman!

**Reply by cschramm** [reply]
Not exactly sure if I get the requirements, but if it's about automatically sending commands or something on (auto-)connect, the `SerialManager` plugin takes a script in its configuration. The script gets passed the device file (I think as its fifth argument; just check what you receive). A shell script with something like `echo "the commands you want to send" > "$5"` should be what you want.

**Reply by NespaLa** [reply]
This SerialManager where to see / find that?

rfcomm -a
rfcomm0: 00:1A:7D:DA:71:13 -> 49:94:11:08:58:C2 channel 7 closed [reuse-dlc release-on-hup ]
rfcomm1: 00:1A:7D:DA:71:13 -> 49:94:11:08:58:C2 channel 7 closed [reuse-dlc release-on-hup ]
rfcomm2: 00:1A:7D:DA:71:13 -> 49:94:11:08:58:C2 channel 7 connected [reuse-dlc release-on-hup tty-attached]
rfcomm49: 00:00:00:00:00:00 channel 1 closed

**Reply by cschramm** [reply]
The plugins item in the tray icon's (right-click) menu brings you to the list and selecting the plugin exposes its configuration.
